### ISSUE TITLE
Error with java wrapper

### ISSUE THREAD

**Issue by kamontat** [skotchpine/asdf-java#44]
The command return ???

```bash
asdf_java_wrapper:2: failed to compile regex: parentheses not balanced
```

**Comment by renatorfr** [comment]
Same here.
I put the wrapper in `.zshrc` file and when I try to set a global version of Java I have this message and the `JAVA_HOME` variable is not set.

**Comment by kamontat** [comment]
Yes, after your zsh file is completed, notify my and I will help you review code. :) Thank you

**Comment by kamontat** [comment]
@skotchpine  I think you should export JAVA_HOME instantly after source wrapper files. 

Currently code, you need to call `asdf` at least 1 time for setting JAVA_HOME env.

**Comment by renatorfr** [comment]
> Thanks for reporting @kamontat.
> 
> What shell are you using?
> Have you run `asdf plugin-update java` recently?
> 
> @renatorfr Silly of me to assume everyone uses bash! I just pushed a sketchy `asdf-java-wrapper.zsh`, can you review my zsh, please?

I worked with zsh. I had to restart my shell to see the `$JAVA_HOME` variable correctly but I saw that there's another issue for that. (Sorry to take so long to answer :))

**Comment by brewster1134** [comment]
i have been using a different solution that accomplishes basically the same thing as the wrapper

```bash
on_change_dir() {
  local java_current
  if java_current=$(asdf current java); then
    # cuts at open parens, and trims any trailing whitespace
    local java_version=$(echo $java_current | cut -d '(' -f 1) | xargs
    export JAVA_HOME=$(asdf where java $java_version)
  else
    echo "No java version set. Type `asdf list-all java` for all versions."
  fi
}
# runs when changing present working directory
function chpwd { on_change_dir }
# runs when new shell session starts
on_change_dir
```

a few differences
* instead of running every time you run asdf, this follows the same principle as asdf and runs each time you change directories
* it has a more elegant way to extract the version from `asdf current` (IMO)
